### ISSUE TITLE
Request: Import Documents into generated code when possible

### ISSUE THREAD

**Issue by MarkKahn** [dotansimha/graphql-code-generator#2350]
Currently in the react templates all gql is imported into the generated code causing them to be defined twice in the final bundle.

My request is to do away with this double definition when possible.  One possible way to go about it is to leave it up to the developer to export the gql from the component where it's initially defined.  If it's exported you can then import from the relative path of the generated code to where the gql is defined.

**Comment by dotansimha** [comment]
@MarkKahn do you mean something like that? https://github.com/dotansimha/graphql-code-generator/pull/2331 
@eddeee888

**Comment by eddeee888** [comment]
@MarkKahn  I'm guessing you mean something like this:

```ts
// QueryA.ts
import gql from 'graphql-tag';

export const QueryA = gql`
  query QueryA {
    project {
      id
      status
    }
  }
`;
```

```ts
// QueryA.generated.tsx
import * as Operations from './QueryA.ts'

export const QueryAComponent = (
  props: QueryAComponentComponentProps
) => (
  <ApolloReactComponents.Query<QueryAComponentQuery, QueryAComponentQueryVariables>
    query={Operations.QueryA}
    {...props}
  />
);
```

There are 2 config values that should help with that: `external` and `importDocumentNodeExternallyFrom`. For example:

```
schema: 'schema.graphql'
generates:
  src/QueryA.generated.tsx:
  documents: './src/**/*.{ts,tsx}' // NOTE: the document scanner fails here...
    plugins:
      - typescript
      - typescript-operations
      - typescript-react-apollo
    config:
      documentMode: external
      importDocumentNodeExternallyFrom: './QueryA'
```

With that said, there's a problem currently where I cannot use [the document scanner](https://graphql-code-generator.com/docs/getting-started/documents-field#document-scanner) to load the `.ts` file...

Maybe I'm doing something wrong @dotansimha ? Here's a codesandbox to test https://codesandbox.io/s/gcg-external-from-non-graphql-file-xe7pl. Run `yarn gg` in the console and we'll get `...import error: Unexpected identifier`

**Comment by dotansimha** [comment]
@eddeee888 Update to latest, we fixed that :) `1.5.1-alpha-2a70e5d7.58` should work (I think you need to add a dependency for `typescript` as well).

**Comment by dotansimha** [comment]
You can do it now, as @eddeee888 explained.
